### PR TITLE
Refactor moab

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made? 🤔
Make it clearer which operations are being performed on moab on storage (vs moab in catalog).

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test preassembly_image_accessioning_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/preassembly_image_accessioning_spec.rb) against stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

File: app/services/complete_moab_service/base.rb [hunk 1]
```diff
@@ -60,12 +60,12 @@ def status_handler
       @status_handler ||= StatusHandler.new(audit_results: results, complete_moab: complete_moab)
     end
 
-    def moab_validator
-      @moab_validator ||= MoabValidator.new(moab: moab, audit_results: results)
+    def moab_on_storage_validator
```
Comment (ndushay) [inline app/services/complete_moab_service/base.rb:63]: great as a method name.

File: app/services/moab_on_storage/storage_directory.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'find'
+
```
Comment (ndushay) [inline app/services/moab_on_storage/storage_directory.rb:4]: add comment:  "distinguish between the moab record in the db and the Moab on local storage (and the zipped moab versions replicated to the cloud)"  ?

File: app/services/moab_on_storage/storage_services_wrapper.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
```
Comment (ndushay) [inline app/services/moab_on_storage/storage_services_wrapper.rb:2]: add comment:  "distinguish between the moab record in the db and the Moab on local storage (and the zipped moab versions replicated to the cloud)"  ?

File: app/services/moab_on_storage/validator.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
```
Comment (ndushay) [inline app/services/moab_on_storage/validator.rb:2]: add comment:  "distinguish between the moab record in the db and the Moab on local storage (and the zipped moab versions replicated to the cloud)"  ?

File: app/services/checksum_validator.rb [hunk 5]
```diff
@@ -35,57 +35,57 @@ def validate_checksums
 
   # @return [Moab::StorageObjectVersion]
   def latest_moab_storage_object_version
-    @latest_moab_storage_object_version ||= moab.version_list.last
+    @latest_moab_storage_object_version ||= moab_on_storage.version_list.last
   end
 
   # @return [Boolean] false if the moab exists, true otherwise
-  def moab_absent?
+  def moab_on_storage_absent?
```
Comment (ndushay) [inline app/services/checksum_validator.rb:42]: 👍🏻

File: app/services/audit/catalog_to_moab.rb [hunk 6]
```diff
@@ -33,18 +33,18 @@ def check_catalog_version
...
 
-      return report_results! unless moab_validator.can_validate_current_comp_moab_status?(complete_moab: complete_moab)
+      return report_results! unless moab_on_storage_validator.can_validate_current_comp_moab_status?(complete_moab: complete_moab)
 
       compare_version_and_take_action
     end
 
-    def moab
-      @moab ||= MoabUtils.moab(druid: druid, storage_location: storage_location)
+    def moab_on_storage
```
Comment (ndushay) [inline app/services/audit/catalog_to_moab.rb:41]: 👍🏻

File: app/services/complete_moab_service/create.rb [hunk 7]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 module CompleteMoabService
-  # Creates CompletedMoab and associated objects based on a moab on disk.
+  # Creates CompletedMoab and associated objects based on a moab on storage.
```
Comment (ndushay) [inline app/services/complete_moab_service/create.rb:4]: change this comment in other similar classes -- maybe do a search on "disk" in codebase  (missed in create_after_validation)

File: app/services/moab_on_storage.rb [hunk 8]
```diff
@@ -1,8 +1,7 @@
 # frozen_string_literal: true
 
-##
-# Utility methods for Moab::StorageObject and Druid
-class MoabUtils
+# Services for a Moab on storage (as opposed to a moab in the catalog)
```
Comment (ndushay) [inline app/services/moab_on_storage.rb:3]: "catalog" --> db ?

or some variant of:
"distinguish between the moab record in the db and the Moab on local storage (and the zipped moab versions replicated to the cloud)" 

?

**Review by ndushay — APPROVED** [review]
Some comments for your consideration ... also check inline as there are some there you might be inclined to do.

So it seems like this PR:

creates the MoabOnStorage namespace in the services directory, and moves classes there:
- models/moab_storage_directory -->   services/moab_on_storage/storage_directory
- services/moab_storage_service -->  services/moab_on_storage/storage_service_wrapper
- services/moab_validator -->  services/moab_on_storage/validator
- services/moab_utils --> services/moab_on_storage.rb  

Would you consider, instead of `MoabOnStorage` as a namespace ... something like `LocalStorage` or `OnPremStorage` ?  Nothing I'll go to the mat over;  just reacting to the verbosity, as I did in storytime yesterday

- models/moab_storage_directory -->   services/local_storage/storage_directory or even /directory
- services/moab_storage_service -->  services/local_storage/storage_service_wrapper
- services/moab_validator -->  services/local_storage/moab_validator
- services/moab_utils --> services/moab_on_storage.rb  -- keep this

Also recommend namespace comment for "LocalStorage" or "MoabOnStorage" is to distinguish between the moab record in the db and the Moab on local storage (and the zipped moab versions replicated to the cloud)"

**Comment by justinlittman** [comment]
@ndushay To consider your naming suggestion, I went back to sections of the code that used the renamed classes and read with both options. (That is, I evaluated based on code read rather than name length.) I think the current naming reads better.

That being said, I don't feel incredibly strongly on this and if you or anyone else do, I'm happy to change.

Making other suggested changes.

**Pull request merged**